src/gpuarray_reduction.c

```diff
@@ -256,113 +801,492 @@ static int   maxandargmaxCheckargs              (maxandargmax_ctx*  ctx){
 	ctx->nds = ctx->src->nd;
 	ctx->ndr = ctx->reduxLen;
 	ctx->ndd = ctx->nds - ctx->ndr;
+	strb_ensure(&ctx->s, 5*1024);
+
 
-	return ctx->ret;
+
+	return reduxSelectModel(ctx);
```

Comment:
I know that it is convenient to do tail calls everywhere like this, but C doesn't have tail-call optimization (or at least not all the time).

It also feels inappropriate that calling reduxCheckArgs ends up compiling the reduction kernel and calling it.